src/TraceEvent/Computers/TraceManagedProcess.cs

```diff
@@ -1230,6 +1283,47 @@ internal static void SetupCallbacks(TraceEventDispatcher source)
             }
         }
 
+        private static string GetTypeName(
+            Address typeID,
+            Dictionary<Address, string> typeNameCache,
+            Dictionary<Address, GCBulkTypeValues> typeMapping)
+        {
+            string typeName;
+            if (typeNameCache.TryGetValue(typeID, out typeName))
+                return typeName;
+
+            GCBulkTypeValues bulkTypeValues;
+            if (typeMapping.TryGetValue(typeID, out bulkTypeValues))
+            {
+                typeName = bulkTypeValues.TypeName;
+                var parameterCount = bulkTypeValues.TypeParameterCount;
+                if (parameterCount > 0)
```

Comment:
You have basically uncovered a bug in the GCBulkType event.    However there is a larger issue of what the proper format for type names is.  format (1) is what the runtime uses internally and for reflection (it is independent of any particular language), and (2) is of course the C# syntax for it.    

Now the question is what should PerfView display?   I would argue that displaying the .NET reflection syntax will just confuse most people.    On the other hand PerfView mostly just displays strings that it was given and thus .NET Reflection syntax is basically the standard used today in PerfView (you will see it in overloads and other things).   Thus we have a policy question (do we care more about being pretty or being consistent?) 

My recommendation is the following: start with the easiest thing (that is fix the bug, and make it easy to do policy changes later).    Handle the policy question as a separate item.  This argues for not displaying C# conventions that is go with (1), at least in the short term. 

The way to do this matter.  We should fix it so that we fix the bug as early as possible, and in a way that allows us to fix the runtime events and not break.    

I believe that GCBulkTImeValue.TypeName should not have generic parameters but should otherwise conform to .NET runtime standards (it does not try to demangle the name). Thus it should return System.WeakReference`1.   We should fix this property so that if it notices a bug, it fixes it (if you have generic parameters it strips the [...] suffix.  

We should create a method 'FullTypeName' that generates (1).   This should be documented to return the .NET Runtime convention for types.   

Optionally we could create a CSharpFullTypeName that generates (2), but we only need this if we change policy.   For now we can skip it.    Logically this is a helper method (users could have defined it, but it is likely to be useful to many clients so we should provide it).     

We should log a bug in .NET Core to fix the GCBulkType event.  

Note that all of this is (logically) independent of where the ID->Name mapping is done.